### ISSUE TITLE
Enum type not assignable to complete union of its literal members

### ISSUE THREAD

**Issue by ericmarkmartin** [microsoft/pyright#11169]
**Describe the bug**
According to the [typing spec for enums](https://typing.python.org/en/latest/spec/enums.html#enum-literal-expansion),

> a type checker should treat a complete union of all literal members as equivalent to the enum type, but not from the enum type to the union.

**Code or Screenshots**

```python
from enum import Enum
from typing import Literal

class Color(Enum):
    RED = "red"
    GREEN = "green"
    BLUE = "blue"

def contraction_id(c: Literal[Color.RED, Color.GREEN, Color.BLUE]:
    return x  # no errors

def expansion_id(c: Color) -> Literal[Color.RED, Color.GREEN, Color.BLUE]):
    return x  # error here
```

The error reported is

```
Type "Color" is not assignable to return type "Literal[Color.RED, Color.GREEN, Color.BLUE]"
  Type "Color" is not assignable to type "Literal[Color.RED, Color.GREEN, Color.BLUE]"
    "Color" is not assignable to type "Literal[Color.RED]"
    "Color" is not assignable to type "Literal[Color.GREEN]"
    "Color" is not assignable to type "Literal[Color.BLUE]"  (reportReturnType)

```

**VS Code extension or command-line**

[pyright playground](https://pyright-play.net/?code=GYJw9gtgBApgdgV2gSwgBzCALlAooiAKFEiiwE81k4BzKVDbKAGWSxhAEMAbQwgY26cAzsKgBhMN0wAKfEgCUALkJQ1UAEq4AIlAC8UAEQgYAE0Or1AcS24AcvqM0T8C%2BqgAhZgFVcjwwBG3AgwFoSmMMBQ-GBwWFz8WMixAPrIpjIAHkosbBw8ANqS0iAAdFraADQSUpilNrj21cV1Xr4AugpQALQAfDUlKu4mWAggcFCZfBFRMJlonHDCyXBpGdkDmF19uexc3EW1ZRXNR-W2dqclpW247UPqI2MTU0A)

**Comment by erictraut** [comment]
Your sample has syntax errors in it. Here's a corrected snippet:

```python
from enum import Enum
from typing import Literal

class Color(Enum):
    RED = "red"
    GREEN = "green"
    BLUE = "blue"

def contraction_id(c: Literal[Color.RED, Color.GREEN, Color.BLUE]) -> Color:
    return c

def expansion_id(c: Color) -> Literal[Color.RED, Color.GREEN, Color.BLUE]:
    return c
```

**Comment by ericmarkmartin** [comment]
> Your sample has syntax errors in it.

Apologies, I was inputting this from my phone since I didn't have a personal machine accessible. I'll update my snippet to yours
